pkg/policy/cell/policy_importer.go

```diff
@@ -0,0 +1,401 @@
+// CIDR identities.
+func (i *policyImporter) processUpdates(ctx context.Context, updates []*policytypes.PolicyUpdate) {
+	if len(updates) == 0 {
+		return
+	}
+
+	i.log.Info("Processing policy updates", logfields.Count, len(updates))
+
+	// First, allocate local identities for all prefixes referenced by policies.
+	//
+	// This must happen before the policies are applied to the endpoints. Doing
+	// so prevents traffic from being dropped. Consider a policy that allows
+	// access to 1.1.1.1/32. It only allows traffic to identities that match that selector.
+	// If we were to apply the policy to the endpoint before allocating an identity for
+	// that prefix, traffic to 1.1.1.1/32 may have the world identity, which would be dropped.
+	//
+	// So, we must always perform identity allocation first, then update policy.
+	// The ony exception is if we are starting up, in which case we may proceed.
+	oldPrefixes := i.updatePrefixes(ctx, updates)
+
+	// Apply changes to the repository.
+	//
+	// As we commit to the policy repository, we must also determine the set of identities
+	// to regenerate here. Identities selected by either outgoing or incoming rules
+	// will have to be regenerated.
+	idsToRegen := &set.Set[identity.NumericIdentity]{}
+	startRevision := i.repo.GetRevision()
+	endRevision := startRevision
+	for _, upd := range updates {
+		var regen *set.Set[identity.NumericIdentity]
+
+		// The standard case: we have an owning resource, either a k8s object
+		// or a file on disk.
+		if upd.Resource != "" {
+			regen, endRevision = i.repo.ReplaceByResource(upd.Rules, upd.Resource)
+		} else {
+			// otherwise, this is a local API call, and we are replacing by labels.
+			var removeLabels []labels.LabelArray
+
+			if len(upd.ReplaceWithLabels) > 0 {
+				removeLabels = []labels.LabelArray{upd.ReplaceWithLabels}
+			} else if upd.ReplaceByLabels {
+				for _, rule := range upd.Rules {
+					removeLabels = append(removeLabels, rule.Labels)
+				}
+			}
+
+			regen, endRevision = i.repo.ReplaceByLabels(upd.Rules, removeLabels)
+		}
+
+		if len(upd.Rules) == 0 {
+			i.log.Info("Deleted policy from repository",
+				logfields.Resource, upd.Resource,
+				logfields.PolicyRevision, endRevision,
+				logfields.Identity, truncate(regen.AsSlice(), 100))
+		} else {
+			i.log.Info("Upserted policy to repository",
+				logfields.Resource, upd.Resource,
+				logfields.PolicyRevision, endRevision,
+				logfields.Identity, truncate(regen.AsSlice(), 100))
+
+		}
+
+		idsToRegen.Merge(*regen)
+
+		// Report that the policy has been inserted in to the repository.
+		// Paranoia: only do so if this will never block.
+		if upd.DoneChan != nil {
+			select {
+			case upd.DoneChan <- endRevision:
+			default:
+				// should never happen
+				i.log.Warn("BUG: failed to deliver policy apply update notification")
+			}
+		}
+
+		// Send a policy update notification
+		if i.monitorAgent != nil && len(upd.Rules) > 0 {
```

Comment:
So we do not send a notification for policy deletes?

Reply:
Crap, that was broken (and I broke it) in `main`. We used to send the policy labels deleted, but we don't have access to that anymore. Hm.

Reply:
OK, I synthesized some labels for the resource case. In main, this previously tagged it with no labels at all.